plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -617,6 +620,23 @@ private function build_query(): void {
 		$groupby = $this->groupby ? 'GROUP BY ' . implode( ', ', (array) $this->groupby ) : '';
 
 		$this->sql = "SELECT $found_rows DISTINCT $fields FROM $orders_table $join WHERE $where $groupby $orderby $limits";
+		$this->build_count_query( $fields, $join, $where, $groupby );
+	}
+
+	/**
+	 * Build SQL query for counting total number of results.
+	 *
+	 * @param string $fields Prepared fields for SELECT clause.
+	 * @param string $join Prepared JOIN clause.
+	 * @param string $where Prepared WHERE clause.
+	 * @param string $groupby Prepared GROUP BY clause.
+	 */
+	private function build_count_query( $fields, $join, $where, $groupby ) {
+		if ( ! isset( $this->sql ) || '' === $this->sql ) {
+			wc_doing_it_wrong( __FUNCTION__, 'Count query can only be build after main query is built.', '7.2.0' );
```

Comment:
Version number needs to be changed here:

```suggestion
			wc_doing_it_wrong( __FUNCTION__, 'Count query can only be build after main query is built.', '7.3.0' );
```